### PR TITLE
Improve code quality

### PR DESCRIPTION
**Pull request by yleeyilin**
Improve code quality by adding assert statements. 
I also helped to standardise commenting and code separations.

### CODE REVIEW

**Review by Joshy837 — APPROVED** [review]
LGTM

**Review by chiageng — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/EditStaffCommandParser.java [hunk 1]
```diff
@@ -94,25 +96,30 @@ public EditStaffCommand parse(String args) throws ParseException {
      * @throws ParseException Indicates the invalid format that users might have entered.
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/EditStaffCommandParser.java:96]: same comment here as earlier; might want to rephrase for standardisation purposes

File: src/main/java/seedu/address/logic/parser/EditSupplierCommandParser.java [hunk 2]
```diff
@@ -93,26 +95,30 @@ public EditSupplierCommand parse(String args) throws ParseException {
      * @throws ParseException Indicates the invalid format that users might have entered.
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/EditSupplierCommandParser.java:95]: same comment here as earlier; might want to rephrase for standardisation purposes

File: src/main/java/seedu/address/logic/parser/HelpCommandParser.java [hunk 3]
```diff
@@ -20,24 +20,23 @@ public class HelpCommandParser implements Parser<HelpCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the HelpCommand
      * and returns a HelpCommand object for execution. Parameter args cannot be null.
+     *
      * @throws ParseException if the user input does not conform the expected format
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/HelpCommandParser.java:24]: same comment here on missing preposition "to"

File: src/main/java/seedu/address/logic/parser/HelpCommandParser.java [hunk 4]
```diff
@@ -20,24 +20,23 @@ public class HelpCommandParser implements Parser<HelpCommand> {
     /**
      * Parses the given {@code String} of arguments in the context of the HelpCommand
      * and returns a HelpCommand object for execution. Parameter args cannot be null.
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/HelpCommandParser.java:22]: same comment here on `args`

File: src/main/java/seedu/address/logic/parser/PinCommandParser.java [hunk 5]
```diff
@@ -16,24 +16,26 @@ public class PinCommandParser implements Parser<PinCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the PinCommand
-     * and returns an PinCommand object for execution.
+     * and returns an PinCommand object for execution. Parameter args cannot be null.
      * @throws ParseException if the user input does not conform the expected format
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/PinCommandParser.java:20]: same comment here on missing preposition "to"

File: src/main/java/seedu/address/logic/parser/RateCommandParser.java [hunk 6]
```diff
@@ -26,28 +26,30 @@ public class RateCommandParser implements Parser<RateCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the RateCommand
-     * and returns a RateCommand object for execution.
+     * and returns a RateCommand object for execution. Parameter args cannot be null.
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/RateCommandParser.java:29]: same comment here on `args`

File: src/main/java/seedu/address/logic/parser/RateCommandParser.java [hunk 7]
```diff
@@ -26,28 +26,30 @@ public class RateCommandParser implements Parser<RateCommand> {
...
+     *
      * @throws ParseException if the user input does not conform the expected format
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/RateCommandParser.java:31]: same comment here on missing preposition "to"

File: src/main/java/seedu/address/logic/parser/RateCommandParser.java [hunk 8]
```diff
@@ -26,28 +26,30 @@ public class RateCommandParser implements Parser<RateCommand> {
...
      */
     public RateCommand parse(String args) throws ParseException {
-        assert (args != null) : "`argument` to pass for rate command is null";
+        assert (args != null) : "argument to pass for rate command is null";
+
         logger.log(Level.INFO, "Going to start parsing for rate command.");
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_RATING);
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/RateCommandParser.java:37]: should have empty line before this to separate from logging code block

File: src/main/java/seedu/address/logic/parser/SearchCommandParser.java [hunk 9]
```diff
@@ -17,34 +17,30 @@ public class SearchCommandParser implements Parser<SearchCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the SearchCommand
-     * and returns a SearchCommand object for execution.
+     * and returns a SearchCommand object for execution. Parameter args cannot be null.
      * @throws ParseException if the user input does not conform the expected format
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/SearchCommandParser.java:21]: same comment here on missing preposition "to"

File: src/main/java/seedu/address/logic/parser/UnpinCommandParser.java [hunk 10]
```diff
@@ -10,28 +10,31 @@
...
- * Parses input arguments and creates a new PinCommand object
+ * Parses input arguments and creates a new UnpinCommand object
  */
 public class UnpinCommandParser implements Parser<UnpinCommand> {
 
     /**
-     * Parses the given {@code String} of arguments in the context of the PinCommand
-     * and returns an PinCommand object for execution.
+     * Parses the given {@code String} of arguments in the context of the UnpinCommand
+     * and returns an UnpinCommand object for execution. Parameter args cannot be null.
```
Comment (jamessinmaojun) [inline src/main/java/seedu/address/logic/parser/UnpinCommandParser.java:19]: same comment here on `args`

**Review by jamessinmaojun — APPROVED** [review]
Just a few comments from me, otherwise lgtm!

1. There are many repeated comments and I have tried my best to comment at all occurrences of the repeated errors, however I may have missed a few so please check through all again.
2. `requireNonNull` seems to be used in conjunction with `assert` for a few commands, but are missing for other commands. Might want to standardise this.

**Pull request merged**